Spigot-Server-Patches/0450-Add-PlayerAttackEntityCooldownResetEvent.patch

```diff
@@ -0,0 +1,37 @@
+From 31dc251b27488c93d817655c6913c69e61f9486b Mon Sep 17 00:00:00 2001
+From: nossr50 <nossr50@gmail.com>
+Date: Thu, 26 Mar 2020 19:44:50 -0700
+Subject: [PATCH] Add PlayerAttackEntityCooldownResetEvent
+
+
+diff --git a/src/main/java/net/minecraft/server/EntityHuman.java b/src/main/java/net/minecraft/server/EntityHuman.java
+index 584166f22..768d724be 100644
+--- a/src/main/java/net/minecraft/server/EntityHuman.java
++++ b/src/main/java/net/minecraft/server/EntityHuman.java
+@@ -1,5 +1,6 @@
+ package net.minecraft.server;
+ 
++import com.destroystokyo.paper.event.player.PlayerAttackEntityCooldownResetEvent;
+ import com.google.common.collect.ImmutableMap;
+ import com.google.common.collect.Lists;
+ import com.mojang.authlib.GameProfile;
+@@ -1057,7 +1058,15 @@ public abstract class EntityHuman extends EntityLiving {
+ 
+                 f *= 0.2F + f2 * f2 * 0.8F;
+                 f1 *= f2;
+-                this.ey();
++                //Paper start - PlayerAttackEntityCooldownResetEvent
++                PlayerAttackEntityCooldownResetEvent playerAttackEntityCooldownResetEvent
++                    = new PlayerAttackEntityCooldownResetEvent((Player) this.getBukkitEntity(), entity.getBukkitEntity(), this.getCooledAttackStrength(0F));
++                playerAttackEntityCooldownResetEvent.callEvent();
++
++                if(!playerAttackEntityCooldownResetEvent.isCancelled()) {
++                        this.ey(); //reset it like normal
```

Comment:
Should probably use an OBFHELPER

Reply:
in this case it's fine as the diff moves the method

Reply:
I didn't know whether or not I should use the obfhelper, sounds like I should have

Reply:
I fixed it, and in the process discovered I might be the most dyslexic person in the world.